File: 3248/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(finalPositionOfSnake(3, []string{"DOWN", "RIGHT", "UP"}))
	fmt.Println(finalPositionOfSnake(2, []string{"RIGHT", "DOWN"}))
}

func finalPositionOfSnake(n int, commands []string) int {
	res := 0
	for _, c := range commands {
		if c == "RIGHT" {
			res++
		} else if c == "LEFT" {
			res--
		} else if c == "DOWN" {
			res += n
		} else {
			res -=n
		}
	}

	return res
}

File: 3248/main.rs
struct Solution {}

impl Solution {
    pub fn final_position_of_snake(n: i32, commands: Vec<String>) -> i32 {
        let mut res = 0;
        for c in commands.iter() {
            if c == "RIGHT" {
                res+=1;
            } else if c == "LEFT" {
                res-=1;
            } else if c == "DOWN" {
                res += n;
            } else {
                res -=n;
            }
        }
        return res;
    }
}

fn main() {
    println!("{}", Solution::final_position_of_snake(2, vec!["RIGHT".to_string(), "DOWN".to_string()]));
}